api/src/main/java/io/opencensus/trace/export/SpanData.java

```diff
@@ -82,11 +87,30 @@ public static SpanData create(
       Timestamp startTimestamp,
       Attributes attributes,
       TimedEvents<Annotation> annotations,
-      TimedEvents<NetworkEvent> networkEvents,
+      TimedEvents<? extends BaseMessageEvent> messageOrNetworkEvents,
```

Comment:
Why do we not deprecate this create and add a new one which accepts MessageEvent and completely remove the BaseMessageEvent which becomes a completely internal thing?

Reply:
This method can't be overloaded with parameter types that differ only in the type of the TimedEvent, because the signatures are the same after generic type erasure.  BaseMessageEvent allows the same method to accept both `TimedEvents<NetworkEvent>` and `TimedEvents<MessageEvent>`.

Reply:
Sure, but then can BaseMessageEvent be package protected?

Reply:
BaseMessageEvent is part of the SpanData API, so I think it should be public.

Reply:
Plus, `SpanData` is not in the same package so it has to be public for `SpanData` to access it.

Reply:
Sure. Will do.